pkg/controller/clusterclaim/clusterclaim_controller.go

```diff
@@ -308,6 +308,7 @@ func (r *ReconcileClusterClaim) reconcileForDeletedCluster(claim *hivev1.Cluster
 func (r *ReconcileClusterClaim) reconcileForNewAssignment(claim *hivev1.ClusterClaim, cd *hivev1.ClusterDeployment, logger log.FieldLogger) (reconcile.Result, error) {
 	logger.Info("cluster assigned to claim")
 	cd.Spec.ClusterPoolRef.ClaimName = claim.Name
+	cd.Spec.PowerState = hivev1.RunningClusterPowerState
```

Comment:
Feels like this should be an option and not the default. Leaving them running would give you immediate response to claims, which might be what some people would want. Hibernation is AWS only right now and probably other clouds will just disregard the setting but arguably validation should be rejecting attempts to create / update powerstate on gcp/azure/etc.

I'd pitch we put this onto the pool spec and make you opt in.

Reply:
I could see making this opt out, but I think the default behavior should be to hibernate. In my view, the cluster pools shine brightest with hibernation. If we add validation to block setting the power state on unsupported platforms, then the clusterpool controller can selectively set the power state based on the platform.

Another option would be to have a desired pool size and a desired hot ready size. The hot ready size would be the number of clusters that are ready and powered on. Let's say a pool has a desired size of 10 and a hot ready size of 3. In steady state, the pool would have 7 hibernating clusters and 3 powered-on clusters. When a claim comes in, the cluster pool will create a new cluster and power on one of the hibernating clusters.